File: nexus-maliopt/src/main/java/com/nexus/modules/textures/ASTCTextureRegistry.java
package com.nexus.modules.textures;

import java.util.concurrent.ConcurrentHashMap;

public class ASTCTextureRegistry {
    private static final ConcurrentHashMap<String, byte[]> astcFiles = new ConcurrentHashMap<>();

    public static void put(String path, byte[] data) {
        astcFiles.put(path, data);
    }

    public static byte[] getASTCData(String path) {
        return astcFiles.get(path);
    }

    public static boolean hasASTCTextures() {
        return !astcFiles.isEmpty();
    }

    public static int count() {
        return astcFiles.size();
    }

    public static void clear() {
        astcFiles.clear();
    }
}

File: nexus-textures/src/main/kotlin/com/nexus/modules/textures/ASTCTextureRegistry.kt
package com.nexus.modules.textures

import net.minecraft.util.Identifier
import java.util.concurrent.ConcurrentHashMap

object ASTCTextureRegistry {
    private val astcFiles = ConcurrentHashMap<Identifier, ByteArray>()

    fun updateMap(resources: Map<Identifier, java.io.InputStream>) {
        astcFiles.clear()
        resources.forEach { (id, input) ->
            val bytes = input.readBytes()
            astcFiles[id] = bytes
        }
    }

    fun getASTCData(id: Identifier): ByteArray? {
        // Tenta id com extensão .astc
        val astcId = Identifier(id.namespace, id.path.replace(".png", ".astc"))
        return astcFiles[astcId]
    }
}
